### PR TITLE
Update rectangular_button.dart

### LINKED ISSUE #3: Add button animations (ripple) in RectangularButton
The button should turn grey-ish when touched and then turn back to its original color when it is let go.
This is the basic behavior one would expect any button to have.
The color variable tied to the money activity color should not be changed in this process.

These might be useful:
- https://docs.flutter.dev/release/breaking-changes/buttons
- https://stackoverflow.com/questions/60414398/custom-button-in-flutter

### PR DESCRIPTION
**Pull request by skfareeduddin**
The button color will turn to grey when clicked!

### CODE REVIEW

**Comment by skfareeduddin** [comment]
This is my proposed solution to the issue.

**Comment by Driexus** [comment]
Hello skfareeduddin,

I am sorry for not writing the needed task a bit better. I did not think anyone would try to fix it so thank you for your effort.
Unfortunately this was not the intended behavior. I have updated the task to better reflect the intended behavior. Feel free to continue working on it if you wish.

File: lib/ui/home/buttons/rectangular_button.dart [hunk 1]
```diff
@@ -3,23 +3,39 @@ import 'package:auto_size_text/auto_size_text.dart';
...
+
+class _RectangularButtonState extends State<RectangularButton> {
+  late final Icon? icon = widget.imageKey != null ? Icon(
+    IconsHelper.getIcon(widget.imageKey!),
+    color: Colors.white,
+    size: 45,
   ) : null;
 
-  final BorderRadius borderRadius = BorderRadius.circular(8.0);
+  late Color currentColor;
```
Comment (Driexus) [inline lib/ui/home/buttons/rectangular_button.dart]: This should not be changed. Instead, the button color should turn greyish (keeping its original color though) when clicked and then reverted back to normal when is let go.

**Review by Driexus — CHANGES_REQUESTED** [review]

File: lib/ui/home/buttons/rectangular_button.dart (hunk 1 above)
Comment (skfareeduddin) [reply lib/ui/home/buttons/rectangular_button.dart]: Ok. I'll do the necessary changes.

**Comment by skfareeduddin** [comment]
I updated the code in the patch-1 as you said. Now the button will turn grey when clicked and then turn back to its original color.

File: lib/ui/home/buttons/rectangular_button.dart [hunk 2]
```diff
@@ -15,56 +16,62 @@ class RectangularButton extends StatefulWidget {
...
     super.initState();
-    currentColor = widget.color;
+    buttonColor = widget.color;
   }
 
   void onTap() {
     setState(() {
-      // Here you are changing the color to grey when clicked
-      currentColor = Colors.grey; 
+      buttonColor = Colors.grey; // Change color to grey when clicked
```
Comment (Driexus) [inline lib/ui/home/buttons/rectangular_button.dart:29]: As I mentioned this should never be changed

File: lib/ui/home/buttons/rectangular_button.dart [hunk 3]
```diff
@@ -15,56 +16,62 @@ class RectangularButton extends StatefulWidget {
...
     });
+    
     log("Clicked on ${widget.description}");
+
+    // Delay for a short time to visually see the color change
+    Future.delayed(Duration(milliseconds: 500), () {
```
Comment (Driexus) [inline lib/ui/home/buttons/rectangular_button.dart:35]: Adding delays on triggers is generally a bad practice causing unwanted behavior (when button remains pressed for example)

**Review by Driexus — COMMENTED** [review]
The solution should use ripple effects. I had updated the issue description, please check it out along with the links if you still wish to solve it.

**Pull request closed without merging**